### PR TITLE
Pricing page rework: Decouple card components from the product logic

### PR DESCRIPTION
**Pull request by manzoorwanijk**
#### Proposed Changes

In continuation of #67485, this PR improves the card components to make them unaware of the product-related logic. It will allow us to extract these components and make them re-usable elsewhere.

#### Testing Instructions

- Do any one of these
    * Click on Jetpack Cloud live link below and wait for the redirect to complete, then goto `/pricing?flags=jetpack/pricing-page-rework-v1`
    * or boot up this PR 
        * Run `git fetch && git checkout update/jp-pricing-rework/decouple-card-components`
        * Run `yarn start-jetpack-cloud`
        * Goto [http://jetpack.cloud.localhost:3000/pricing?flags=jetpack/pricing-page-rework-v1](http://jetpack.cloud.localhost:3000/pricing?flags=jetpack/pricing-page-rework-v1)
- Confirm that everything is same as before

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

Related to #

### CODE REVIEW

File: client/my-sites/plans/jetpack-plans/product-store/items-list/all-items.tsx [hunk 1]
```diff
@@ -45,29 +48,46 @@ export const AllItems: React.FC< AllItemsProps > = ( {
...
+						<>
+							{ item?.shortDescription || item.description } <br />
+							{ ! hideMoreInfoLink && (
+								<MoreInfoLink onClick={ onClickMoreInfoFactory( item ) } item={ item } />
+							) }
+						</>
+					);
+
 					return (
 						<SimpleProductCard
```
Comment (jkguidaven) [inline client/my-sites/plans/jetpack-plans/product-store/items-list/all-items.tsx]: Now that this card becomes a generic one. Maybe we can rename it to SimpleItemCard or just SimpleCard.

**Review by jkguidaven — COMMENTED** [review]
Just one minor comment but the idea to decouple the logic is awesome.. thank you for doing this 👍

File: client/my-sites/plans/jetpack-plans/product-store/items-list/all-items.tsx (hunk 1 above)
Comment (manzoorwanijk) [reply client/my-sites/plans/jetpack-plans/product-store/items-list/all-items.tsx]: Oh, that's a good idea. Let me do that.

Comment (manzoorwanijk) [reply client/my-sites/plans/jetpack-plans/product-store/items-list/all-items.tsx]: Done. Thanks

**Review by jkguidaven — APPROVED** [review]
awesome LGTM now 👍 . tested and it is working correctly.

**Pull request merged**